### PR TITLE
Follow XDG Base Directory Specification

### LINKED ISSUE #2: Store "EnhancedDiscord" in either $XDG_DATA_HOME or $XDG_CACHE_HOME (Linux)
From the [XDG Base Directory Specification](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html):
> `$XDG_DATA_HOME` defines the base directory relative to which user specific data files should be stored.

> `$XDG_CACHE_HOME` defines the base directory relative to which user specific non-essential data files should be stored.

Currently "EnhancedDiscord" is extracted to the directory the script is in, which can clutter the file system and in case `LinuxED.py` is installed system‐wide, it may not be allowed at all (e.g., "`FileNotFoundError: [Errno 2] No such file or directory: '/usr/bin/EnhancedDiscord/config.json'`"). The "XDG Base Directory Specification" was made to address these issues.

Depending on whether the resulting "EnhancedDiscord" directory is either a data directory that needs to be revisited continuously (e.g., this is where the ED extension "lives") or if it’s treated more like a cache (it’s downloaded and then gets installed into Discord’s own directory/‐ies) the "EnhancedDiscord" directory should be placed under `$XDG_DATA_HOME` (first case) or `$XDG_CACHE_HOME` (second case).

### PR DESCRIPTION
**Pull request by Freso**
Place EnhancedDiscord files under $XDG_DATA_HOME as specified by the [XDG Base Directory Specification version 0.7](https://specifications.freedesktop.org/basedir-spec/basedir-spec-0.7.html).

This allows LinuxED to be run from anywhere while still having its data files located in a single location and also means for users that there’s one fewer location where data files will be scattered around their file system.

Fixes https://github.com/Cr3atable/LinuxED/issues/2

### CODE REVIEW

**Comment by twnlink** [comment]
I'd prefer for this to be optional.
There's also a merge conflict with the latest commit. I will merge once these two issues are fixed.

**Comment by twnlink** [comment]
This PR is now merged however I would like to comment on how the new update works.
Rather than defaulting to the XDG Base Directory, the installer now provides the option to use it upon starting as shown in the following screenshot:
Example of option

If you use this option the installer will then automatically use the XDG Base Directory.
If there is an EnhancedDiscord installation within the XDG Base Directory on start, the installer will default to the XDG Base directory.
This allows users to optionally pick what directory they want to install EnhancedDiscord in and makes sure not to be too much of a hassle if you choose anything besides the default.

Thank you for your pull request and issue.

File: LinuxED.py [hunk 1]
```diff
@@ -29,22 +29,19 @@
     # Define the starting variables, these are all their own thing.
     username = getpass.getuser()
     dirpath = os.path.realpath('')
+    currentdir = True
+    if 'XDG_DATA_HOME' in os.environ:
+        if os.path.exists(os.path.join(os.environ['XDG_DATA_HOME'], "EnhancedDiscord")):
+            dirpath = os.environ['XDG_DATA_HOME']
+            currentdir = False
+    elif 'HOME' in os.environ:
+        if os.path.exists(os.path.join(os.path.join(os.environ['HOME'], '.local', 'share'), "EnhancedDiscord")):
```
Comment (Freso) [inline LinuxED.py:38]: `os.path.join(os.path.join(…), …)` is superfluous. Just use
```python
         if os.path.exists(os.path.join(os.environ['HOME'], '.local', 'share', "EnhancedDiscord")):
```

**Comment by Freso** [comment]
Thanks. :) I don’t agree with not making the XDG standard approach the default, but I’m not gonna bother you more about that now that it is at least an option. ;)

**Pull request merged**